peer/brontide.go

```diff
@@ -2253,68 +2291,48 @@ func (p *Brontide) queueHandler() {
 // connection is still active.
 //
 // NOTE: This method MUST be run as a goroutine.
-func (p *Brontide) pingHandler() {
-	defer p.wg.Done()
-
-	pingTicker := time.NewTicker(pingInterval)
-	defer pingTicker.Stop()
+func (p *Brontide) pingManagerLiason(
+	pingChan <-chan lnwire.Ping,
+	failChan <-chan struct{},
+) {
 
-	// TODO(roasbeef): make dynamic in order to create fake cover traffic
-	const numPongBytes = 16
-
-	blockEpochs, err := p.cfg.ChainNotifier.RegisterBlockEpochNtfn(nil)
-	if err != nil {
-		p.log.Errorf("unable to establish block epoch "+
-			"subscription: %v", err)
-		return
-	}
-	defer blockEpochs.Cancel()
-
-	var (
-		pingPayload [wire.MaxBlockHeaderPayload]byte
-		blockHeader *wire.BlockHeader
-	)
-out:
+	defer p.wg.Done()
+	msgWritten := make(chan error)
 	for {
 		select {
-		// Each time a new block comes in, we'll copy the raw header
-		// contents over to our ping payload declared above. Over time,
-		// we'll use this to disseminate the latest block header
-		// between all our peers, which can later be used to
-		// cross-check our own view of the network to mitigate various
-		// types of eclipse attacks.
-		case epoch, ok := <-blockEpochs.Epochs:
-			if !ok {
-				p.log.Debugf("block notifications " +
-					"canceled")
-				return
-			}
-
-			blockHeader = epoch.BlockHeader
-			headerBuf := bytes.NewBuffer(pingPayload[0:0])
-			err := blockHeader.Serialize(headerBuf)
+		case ping := <-pingChan:
```

Comment:
I think we can instead pass callbacks to the ping manager. See proposed patch

Reply:
> designed to take a channels first approach to things which as far as I understand is the preferred go model of thinking

Not necessarily. Channels are nice within a subsystem when used in a "closed" way such that how the system is used from the outside doesnt affect things. With the current impl as it currently stands, there is the whole "ping manager liason" thing which imo is leaking implementation into the brontide system. I opt for keeping things contained within the ping manager

> I think it's important to not pollute the Brontide struct with additional state 

where is the polluted state?